src/test/run-make/expand-derive/input_pp.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// minimal junk
+#![feature(no_core)]
+#![no_core]
+
+struct A {
+    a: i32,
+}
+#[automatically_derived]
+#[allow(unused_qualifications)]
+impl ::clone::Clone for A {
+    #[inline]
+    fn clone(&self) -> A {
+        match *self {
+            A { a: ref __self_0_0 } => {
+                ::clone::assert_receiver_is_clone(&(*__self_0_0));
```

Comment:
In other words this is a behavior change, but it is now OKed by accepted rust-lang/rfcs#1521.